test/gtest/coll/test_scatterv.cc

```diff
@@ -0,0 +1,309 @@
+
+                UCC_CHECK(ucc_mc_alloc(&ctxs[r]->src_mc_header,
+                                       ucc_dt_size(dtype) * all_counts,
+                                       mem_type));
+                coll->src.info_v.buffer = ctxs[r]->src_mc_header->addr;
+                UCC_CHECK(ucc_mc_memcpy(coll->src.info_v.buffer,
+                          ctxs[r]->init_buf, ucc_dt_size(dtype) * all_counts,
+                          mem_type, UCC_MEMORY_TYPE_HOST));
+            }
+            if (r != root || !inplace) {
+                UCC_CHECK(ucc_mc_alloc(&ctxs[r]->dst_mc_header,
+                                       ucc_dt_size(dtype) * my_count,
+                                       mem_type));
+                coll->dst.info.buffer = ctxs[r]->dst_mc_header->addr;
+            }
+            if (inplace) {
+                coll->mask |= UCC_COLL_ARGS_FIELD_FLAGS;
+                coll->flags |= UCC_COLL_ARGS_FLAG_IN_PLACE;
+            }
+            if (persistent) {
+                coll->mask |= UCC_COLL_ARGS_FIELD_FLAGS;
+                coll->flags |= UCC_COLL_ARGS_FLAG_PERSISTENT;
+            }
+        }
+    }
+    void data_fini(UccCollCtxVec ctxs)
+    {
+        for (auto r = 0; r < ctxs.size(); r++) {
+            ucc_coll_args_t *coll = ctxs[r]->args;
+            if (r == root) {
+                UCC_CHECK(ucc_mc_free(ctxs[r]->src_mc_header));
+                free(coll->src.info_v.counts);
+                free(coll->src.info_v.displacements);
+            }
+            if (r != root || !inplace) {
+                UCC_CHECK(ucc_mc_free(ctxs[r]->dst_mc_header));
+            }
+            ucc_free(ctxs[r]->init_buf);
+            free(coll);
+            free(ctxs[r]);
+        }
+        ctxs.clear();
+    }
+    void reset(UccCollCtxVec ctxs)
+    {
+        ucc_coll_args_t *coll;
+        ucc_datatype_t   dtype = (ctxs[0])->args->dst.info.datatype;
+        size_t           my_count;
+
+        for (auto r = 0; r < ctxs.size(); r++) {
+            coll = ctxs[r]->args;
+            my_count = coll->dst.info.count;
+            if (r != root || !inplace) {
+                clear_buffer(coll->dst.info.buffer,
+                    my_count * ucc_dt_size(dtype),
+                    mem_type, 0);
+            }
+        }
+    }
+    bool data_validate(UccCollCtxVec ctxs)
+    {
+        bool     ret               = true;
+        int      root              = ctxs[0]->args->root;
+        size_t   dt_size = ucc_dt_size((ctxs[0])->args->dst.info.datatype);
+        size_t   my_count;
+        uint8_t *dsts;
+
+        for (auto r = 0; r < ctxs.size(); r++) {
+            if (r == root && inplace) {
+                continue;
+            }
+            my_count = (ctxs[r])->args->dst.info.count;
+            if (UCC_MEMORY_TYPE_HOST != mem_type) {
+                dsts = (uint8_t *)ucc_malloc(my_count * dt_size, "dsts buf");
+                EXPECT_NE(dsts, nullptr);
```

Comment:
ASSERT_NE

Reply:
changes to ucc_assert as ASSERT_NE can only be used inside void function and this data_validate function returns bool. Same for data_validate in test_gatherv.